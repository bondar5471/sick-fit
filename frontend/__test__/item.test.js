import ItemComponent from '../components/Item'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
  id: 'test',
  title: 'test',
  price: '4000',
  description: 'test description',
  image: 'test.jpg',
  largeImage: 'largetest.js'
}

describe('<Item />', () => {
  // const wrapper = shallow(<ItemComponent item={fakeItem} />)

  // it('render price and title properly', () => {
  //   const PriceTag = wrapper.find('PriceTag')
  //   expect(PriceTag.children().text()).toBe('$50')
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
  
  // })

  // it('render images properly', ()=> {
  //   expect(wrapper.find('img').props().src).toBe(fakeItem.image)
  //   expect(wrapper.find('img').props().alt).toBe(fakeItem.title)
  // })

  // it('renders out the button properly' , () => {
  //   const buttonList = wrapper.find('.buttonList')
  //   expect(buttonList.children()).toHaveLength(3)
  //   expect(buttonList.find('Link')).toHaveLength(1)
  //   expect(buttonList.find('AddToCart')).toHaveLength(1)
  //   expect(buttonList.find('DeleteItem')).toHaveLength(1)
  // })

  it('render and matches the snapshot ', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>)
    expect(toJSON(wrapper)).toMatchSnapshot()
    });
});