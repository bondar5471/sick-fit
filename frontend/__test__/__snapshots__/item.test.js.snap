// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> render and matches the snapshot  1`] = `
<ItemStyles__Item>
  <img
    alt="test"
    src="test.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "test",
          },
        }
      }
    >
      <a>
        test
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    test description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "test",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="test"
    />
    <DeleteItem
      id="test"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
